atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunner.java

```diff
@@ -199,21 +202,56 @@ private SweepResults doRun(TableReference tableRef,
     }
 
     /**
-     * Returns the number of deleted (cell, timestamp) pairs.
+     * Partitions batches of cells down to 2 * deleteBatchSize blocks per sweep deletion call.
+     * Returns the number of blocks that were deleted.
      */
-    private int sweepBatch(TableReference tableRef, List<CellToSweep> batch, RunType runType) {
-        Multimap<Cell, Long> startTimestampsToSweepPerCell = ArrayListMultimap.create();
-        List<Cell> sentinels = Lists.newArrayList();
+    private int sweepBatch(TableReference tableRef, List<CellToSweep> batch, RunType runType,
+            int deleteBatchSize) {
+        int numberOfSweptCells = 0;
+
+        // Previously we've grouped cells to have at least deleteBatchSize blocks per batch. Therefore we expect most
+        // of the batches to be slightly bigger than deleteBatchSize. Partitioning such batches with deleteBatchSize as
+        // a limit make us have a small second batch. Therefore, we set our limit per batch to 2 * deleteBatchSize.
+        deleteBatchSize = 2 * deleteBatchSize;
```

Comment:
(style nit) Mutating function's arguments is a bit counter-intuitive. Maybe you should just pass the doubled number into this function, and call the argument maxDeleteBatchSize